marquez_airflow/dag.py

```diff
@@ -10,330 +10,146 @@
-                          airflow_run_id=dag_run_id,
-                          marquez_namespace=self.marquez_namespace)
-            log.info(f'Successfully recorded job run: {step.name}',
-                     airflow_dag_id=self.dag_id,
-                     airflow_dag_execution_time=start_time,
-                     marquez_run_id=marquez_jobrun_id,
-                     marquez_namespace=self.marquez_namespace,
-                     duration_ms=(self._now_ms() - report_job_start_ms))
-
-        # Store the mapping for all the steps associated with a task
+                # Log error, then ...
+                log.error(
+                    f"""
+                    Failed to collect task metadata: {e}
+                    """,
+                    marquez_namespace=self._marquez_namespace,
+                    duration_ms=(self._now_ms() - start)
+                )
+                continue
+
+        return dag_run
+
+    def _collect_task_meta(self, task):
         try:
-            self._job_id_mapping.set(
-                JobIdMapping.make_key(task_name, dag_run_id),
-                json.dumps(marquez_jobrun_ids))
-
+            # ...
+            extractor = _EXTRACTORS.extractor_for_task(task)
+            task_meta = extractor.extract(task)
+
+            # (1)
+            self._collect_source_meta(task_meta)
+
+            # (2)
+            inputs = []
+            if task_meta.inputs:
+                for input in task_meta.inputs:
+                    self._collect_dataset_meta(input, task_meta.source_name)
+                inputs = list(
+                    map(lambda input: {
+                        'namespace': self._marquez_namespace,
+                        'name': input
+                    }, task_meta.inputs)
+                )
+
+            # (3)
+            outputs = []
+            if task_meta.outputs:
+                for output in task_meta.outputs:
+                    self._collect_dataset_meta(output, task_meta.source_name)
+                outputs = list(
+                    map(lambda output: {
+                        'namespace': self._marquez_namespace,
+                        'name': output
+                    }, task_meta.outputs)
+                )
+
+            # (4)
+            self._marquez_client.create_job(
+                namespace_name=self._marquez_namespace,
+                job_name=task_meta.name,
+                job_type=JobType.BATCH,
+                input_dataset=inputs,
+                output_dataset=outputs,
+                location='https://github.com/' + task_meta.name + '/blob/2294bc15eb49071f38425dc927e48655530a2f2e',
+                context=task_meta.context,
+                description=self.description)
+
+            log.info(
+                f"""
+                Successfully collected metadata for task: {task_meta.name}
+                """,
+                marquez_namespace=self._marquez_namespace
+            )
```

Comment:
As I understand this, we still have ` marquez_client.mark_job_run_as_started` and ` marquez_client.mark_job_run_as_complete` on the TODO list.
I will review when this comes back.